storage/uri.go

```diff
@@ -59,53 +61,131 @@ func (u *uri) String() string {
 	return u.raw
 }
 
+// parentGeneric is a generic function that returns the last element of a
+// path after splitting it on "/". It should be suitable for most URIs.
+func parentGeneric(location string) (string, error) {
+
+	// trim leading forward slashes
+	trimmed := 0
+	for location[0] == '/' {
+		location = location[1:]
+		trimmed++
+
+		// if all we have left is an empty string, than this URI
+		// pointed to a UNIX-style root
+		if len(location) == 0 {
+			return "", URIRootError
+		}
+	}
+
+	components := strings.Split(location, "/")
+
+	if len(components) == 1 {
+		// trimmed <= 2 makes sure we handle UNIX-style paths on
```

Comment:
This comment seems out of date somehow?

Reply:
And if this method is explicitly for handling non-file URIs  why is windows path handling in here?

Reply:
I removed the comment

Reply:
Is that the right fix? If a developer thought a comment was needed then surely there is a wider issue at play?

Reply:
I thought the same, but I couldn't think of a proper comment, since I don't even know why the comment was there in the first place.

Reply:
Yeah, I think this is not a relevant comment anymore. This was part of the logic for handling Windows drive letters.